packages/office-ui-fabric-react/src/components/Button/Button.Props.ts

```diff
@@ -44,15 +45,23 @@ export interface IButtonProps extends React.HTMLProps<HTMLButtonElement | HTMLAn
   text?: string;
 
   /**
+  *
+  */
+  contextualMenuItems?: IContextualMenuItem[];
+
+  /**
    * The button icon shown in command or hero type.
-   * Deprecated at v1.2.3, to be removed at >= v2.0.0. Use IconButton instead.
    */
   icon?: string;
 
   /**
+   * The button icon shown to the right of the text
+   */
+  splitIcon?: string;
```

Comment:
+1.
In VSTS we have the notion of extensibility within our menus where 3rd party extensions can contribute items to menus. In this case, we don't know the menu items up front, and will need to (sometimes) make an AJAX request to pull down the items to show. So, if we had the ability to provide our own component as the menu, that would be very useful.